workers/datasets_based/tests/workers/test_config_names.py

```diff
@@ -0,0 +1,112 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright 2022 The HuggingFace Authors.
+
+from dataclasses import replace
+from http import HTTPStatus
+
+import pytest
+from libcommon.exceptions import CustomError
+from libcommon.simple_cache import DoesNotExist, get_response
+
+from datasets_based.config import AppConfig
+from datasets_based.workers.config_names import ConfigNamesWorker
+
+from ..fixtures.hub import HubDatasets
+
+
+def get_worker(
+    dataset: str,
+    app_config: AppConfig,
+    force: bool = False,
+) -> ConfigNamesWorker:
+    return ConfigNamesWorker(
+        job_info={
+            "type": ConfigNamesWorker.get_job_type(),
+            "dataset": dataset,
+            "config": None,
+            "split": None,
+            "job_id": "job_id",
+            "force": force,
+        },
+        app_config=app_config,
+    )
+
+
+def test_should_skip_job(app_config: AppConfig, hub_public_csv: str) -> None:
+    dataset = hub_public_csv
+    worker = get_worker(dataset, app_config)
+    assert worker.should_skip_job() is False
+    # we add an entry to the cache
+    worker.process()
+    assert worker.should_skip_job() is True
+    worker = get_worker(dataset, app_config, force=True)
+    assert worker.should_skip_job() is False
+
+
+def test_process(app_config: AppConfig, hub_public_csv: str) -> None:
+    dataset = hub_public_csv
+    worker = get_worker(dataset, app_config)
+    assert worker.process() is True
+    cached_response = get_response(kind=worker.processing_step.cache_kind, dataset=hub_public_csv)
```

Comment:
is it using an actual mongo database for the cache when running tests ?

Reply:
Yes:
- we launch `make up` before launching the test, then `make down`
https://github.com/huggingface/datasets-server/blob/e32590450a1f0c33439b41bb5b0b8a44f58fd9db/tools/PythonTest.mk#L4-L7
- which corresponds to launching the docker compose described in the `DOCKER_COMPOSE` variable
https://github.com/huggingface/datasets-server/blob/e32590450a1f0c33439b41bb5b0b8a44f58fd9db/tools/Docker.mk#L1-L7
- which turns to be only a mongo container:
https://github.com/huggingface/datasets-server/blob/e32590450a1f0c33439b41bb5b0b8a44f58fd9db/services/api/Makefile#L6-L7
- see:
https://github.com/huggingface/datasets-server/blob/e32590450a1f0c33439b41bb5b0b8a44f58fd9db/tools/docker-compose-mongo.yml#L1-L10

Every service or worker has a different mongo container with a different exposed port, so that we can run multiple tests at the same time (just in case).

Also: in the tests, we drop the content of the base before each test.